server/server.go

```diff
@@ -1665,17 +1673,56 @@ func (s *Server) UnmarkSnapshotRecovering(ctx context.Context) error {
 	return err
 }
 
+type servicePrimary struct {
```

Comment:
Why do we need this struct?

Reply:
only keep result is unchanged, how about removing error and output it in log if it is not nil?